api/prometheus/v1/remote/remote_headers.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const (
+	versionHeader        = "X-Prometheus-Remote-Write-Version"
+	version1HeaderValue  = "0.1.0"
+	version20HeaderValue = "2.0.0"
+	appProtoContentType  = "application/x-protobuf"
+)
+
+// Compression represents the encoding. Currently remote storage supports only
+// one, but we experiment with more, thus leaving the compression scaffolding
+// for now.
+type Compression string
+
+const (
+	// SnappyBlockCompression represents https://github.com/google/snappy/blob/2c94e11145f0b7b184b831577c93e5a41c4c0346/format_description.txt
+	SnappyBlockCompression Compression = "snappy"
+)
+
+// WriteProtoFullName represents the fully qualified name of the protobuf message
+// to use in Remote write 1.0 and 2.0 protocols.
+// See https://prometheus.io/docs/specs/remote_write_spec_2_0/#protocol.
+type WriteProtoFullName protoreflect.FullName
+
+const (
+	// WriteProtoFullNameV1 represents the `prometheus.WriteRequest` protobuf
+	// message introduced in the https://prometheus.io/docs/specs/remote_write_spec/.
+	// DEPRECATED: Use WriteProtoFullNameV2 instead.
```

Comment:
For receivers who wants to support it for longer, deprecation might cause some friction, but maybe worth it? 🤔